### PR TITLE
Add ufs-utils template

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
### Summary

This PR, in conjunction with spack PR [#300](https://github.com/JCSDA/spack/pull/300) provides stack building support required to build [UFS_UTILS](https://github.com/ufs-community/UFS_UTILS).

### Testing

This PR has been compiled on both the NOAA CSP AWS PW platform and RDHPCS Hera. The resulting stack supports the compilation of the UFS_UTILS applications, including the available unit-tests.

### Applications affected

No applications are impacted. UFS_UTILS is now supported. 

### Systems affected

None.

### Dependencies

- https://github.com/JCSDA/spack/pull/300

### Issue(s) addressed

This PR addresses issue #644.

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by climbfuji** [comment]
@HenryWinterbottom-NOAA I merged the spack PR. The new hash for the spack submodule is https://github.com/JCSDA/spack/commit/24657b60ad3ea50c99a1d68de8c178a410ba54eb. Please:
- update the spack submodule pointer in spack-stack to point to this hash
- pull in the updates from spack-stack develop

Thanks!

**Comment by HenryRWinterbottom** [comment]
@climbfuji Would you like me to create a new branch/issue for this?

**Comment by climbfuji** [comment]
No, you just need to push these updates to this branch

**Comment by HenryRWinterbottom** [comment]
> No, you just need to push these updates to this branch

@climbfuji I made the requested updates. Please let me if I misunderstood and/or missed something.

File: .gitmodules [hunk 1]
```diff
@@ -3,7 +3,7 @@
   #url = https://github.com/spack/spack
   #branch = develop
   url = https://github.com/jcsda/spack
-  branch = jcsda_emc_spack_stack
+  branch = JCSDA/spack@24657b6
```
Comment (climbfuji) [inline .gitmodules]: This change needs to be reverted. You need to update the submodule pointer, not `.gitmodules`.

**Review by climbfuji — CHANGES_REQUESTED** [review]
Let me know if you need my help with this.

File: .gitmodules (hunk 1 above)
Comment (HenryRWinterbottom) [reply .gitmodules]: @climbfuji Can you point me to the correct file and/or give me explicit instructions? I don't know what the "submodule pointer" is referring to.

Comment (climbfuji) [reply .gitmodules]: In your local spack-stack clone of your branch `subfeature/packages_py`:
```
cd spack
git remote update
git checkout origin/jcsda_emc_spack_stack
git log
# this should show you the hash I mentioned at the very top
cd ..
git diff
# you should see a diff showing up for "spack (new commits)"
git add spack
git commit -m "Update submodule pointer for spack"
git push YOUR_REMOTE_NAME subfeature/packages_py
```

Comment (HenryRWinterbottom) [reply .gitmodules]: @climbfuji OK. I just submitted attempt number 2.

Comment (climbfuji) [reply .gitmodules]: submodule pointer is looking good now, just need to revert `.gitmodules`.

Comment (HenryRWinterbottom) [reply .gitmodules]: To save headaches, how should I properly do that given our current state of the repo?

Comment (climbfuji) [inline .gitmodules]:
```suggestion
  branch = jcsda_emc_spack_stack
```

**Review by climbfuji — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 1 above)
Comment (HenryRWinterbottom) [reply .gitmodules]: @climbfuji OK, reverted. 

Thank you for your patience with me.

**Review by climbfuji — APPROVED** [review]

File: .gitmodules (hunk 1 above)
Comment (climbfuji) [reply .gitmodules]: CI tests are running now, thanks!

Comment (climbfuji) [reply .gitmodules]: They shouldn't be affected by your changes, but I'll let them finish nonetheless before merging.

Comment (HenryRWinterbottom) [reply .gitmodules]: @climbfuji No problem at all! I am happy to contribute and thank you again  for the opportunity.

File: configs/templates/ufs-utils/spack.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        lmod: $env/install/modulefiles
+        tcl: $env/install/modulefiles
+
+  view: false
+  include: []
+
+  specs:
+    - ufs-utils-env~debug  
+
+  view: false
```
Comment (climbfuji) [inline configs/templates/ufs-utils/spack.yaml]: The CI unit tests found an error:
```
FAILED tests/test_stack_create.py::test_apps[ufs-utils] - spack.util.spack_yaml.SpackYAMLError: error parsing YAML: near /home/runner/work/spack-stack/spack-stack/configs/templates/ufs-utils/spack.yaml, 1, 2: found duplicate key "view" with value "False" (original value: "False")
```

Please remove lines 20 and 21 (just for the sake of not having unnecessary blank lines), hopefully that will do.

**Review by climbfuji — CHANGES_REQUESTED** [review]

File: configs/templates/ufs-utils/spack.yaml (hunk 2 above)
Comment (HenryRWinterbottom) [reply configs/templates/ufs-utils/spack.yaml]: @climbfuji I just make the correction. Please see my note in the commit.

File: configs/templates/ufs-utils/spack.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  - netcdf-fortran@4.5.3
+  - esmf@8.4.0b08
+  - nco@4.9.1
+  - nccmp@1.8.9.0
+  - libpng@1.6.35
+  - prod-util@1.2.1
+  - jasper@2.0.25
+  - bufr@11.7.0
+  - wgrib2@2.0.8
+  include:
```
Comment (climbfuji) [inline configs/templates/ufs-utils/spack.yaml]: You should remove those includes, they will get added automatically by `spack stack create env`. Basically, your file should look like https://github.com/JCSDA/spack-stack/blob/develop/configs/templates/empty/spack.yaml with the specs section being expanded.

**Review by climbfuji — CHANGES_REQUESTED** [review]

File: configs/templates/ufs-utils/spack.yaml (hunk 3 above)
Comment (HenryRWinterbottom) [reply configs/templates/ufs-utils/spack.yaml]: @climbfuji Done.

**Review by climbfuji — APPROVED** [review]

**Pull request merged**